### PR TITLE
re-added MongoDbUriSupplier in order to parse options

### PR DESCRIPTION
**Pull request by thjaeckle**
from a given/configured/injected MongoDB URI and apply additional options based on that base URI config

### CODE REVIEW

File: services/utils/persistence/src/test/java/org/eclipse/ditto/services/utils/persistence/mongo/config/DefaultOptionsConfigTest.java [hunk 1]
```diff
@@ -49,7 +50,9 @@ public void initMongoDbConfig() {
     public void assertImmutability() {
         assertInstancesOf(DefaultOptionsConfig.class,
                 areImmutable(),
-                provided(ReadPreference.class, ReadConcern.class, WriteConcern.class).areAlsoImmutable());
+                provided(ReadPreference.class, ReadConcern.class, WriteConcern.class).areAlsoImmutable(),
+                assumingFields("extraUriOptions")
+                        .areSafelyCopiedUnmodifiableCollectionsWithImmutableElements());
```
Comment (Yannic92) [inline services/utils/persistence/src/test/java/org/eclipse/ditto/services/utils/persistence/mongo/config/DefaultOptionsConfigTest.java:55]: This assumption seems not to be correct. The map stored in DefaultOptionsConfig seems to be modfiable (and is provided by a getter)

**Review by Yannic92 — CHANGES_REQUESTED** [review]

File: services/utils/persistence/src/test/java/org/eclipse/ditto/services/utils/persistence/mongo/config/DefaultOptionsConfigTest.java (hunk 1 above)
Comment (thjaeckle) [reply services/utils/persistence/src/test/java/org/eclipse/ditto/services/utils/persistence/mongo/config/DefaultOptionsConfigTest.java:55]: You're right 👍

**Comment by thjaeckle** [comment]
@Yannic92 I made the map really immutable now, could you have another look?

**Review by Yannic92 — APPROVED** [review]
LGTM

**Pull request merged**